components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/TaskDataBase.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.ntask.coordination.task;
+
+import org.wso2.micro.integrator.ntask.coordination.task.db.connector.CoordinatedTaskDBConnector;
+
+import java.sql.SQLException;
+import java.util.List;
+import javax.sql.DataSource;
+
+/**
+ * The layer which connects to the task data base.
+ */
+public class TaskDataBase {
+
+    /**
+     * Connector for the data base.
+     */
+    private CoordinatedTaskDBConnector coordinatedTaskDBConnector;
+
+    public TaskDataBase(DataSource dataSource) throws SQLException {
```

Comment:
Can we create an Exception wrapping the SQLException for that TaskDBCleaner doesn't have to know about the underlying sql implementation. Basically, if we decide to replace the data layer with some other type (file, nosql, etc.) we wouldn't have to change any code in TaskDBCleaner.

Reply:
This wrappe exception needs to be thrown by the CoordinatedTaskDBConnector itself.